app/views/admin/products_v3/_edit_image.html.haml

```diff
@@ -0,0 +1,15 @@
+= render ModalComponent.new id: "#modal_edit_product_image_#{image.id}", instant: true do
+  %h2= t(".title")
+
+  %p= image_tag image.variant(:product)
+
+  -# Submit to controller, because StimulusReflex doesn't support file uploads
+  = form_for [:admin, product, image], url: admin_product_image_path(product, image),
+      html: { multipart: true } do |f|
+    %input{ type: :hidden, name: :return_url, value: return_url}
+    = f.hidden_field :viewable_id, value: product.id
+
+    %p
+      = f.label :attachment, t(".upload"), class: "button primary icon-upload-alt"
+      = f.file_field :attachment, style: "display: none"
+      = f.submit # todo: submit automatically
```

Comment:
I think that the future will be to upload the files directly to the storage (AWS) from the browser. Rails has everything we need for that. The form submission then just contains the id of the uploaded file and a checksum and should work with StimulusReflex. The downside of this approach is probably that the storage validations gem doesn't work with this. It inspects the uploaded files and we would need to download it from the storage first. Still possible. We need to download it anyway to generate all the variants.